Makefile

```diff
@@ -62,10 +62,13 @@ perftest: compile_all
 genclean:
 	rm libsharp/sharp_legendre.c || exit 0
 
-python/libsharp/libsharp.so: python/libsharp/libsharp.pyx $(LIB_libsharp)
+python/libsharp/libsharp.so: python/libsharp/libsharp.pyx $(all_lib)
+ifndef PIC_CFLAGS
+	$(error Python extension must be built using the --enable-pic configure option.)
+endif
 	cython $<
-	$(CC) -fPIC `python-config --cflags` -I$(INCDIR) -o python/libsharp/libsharp.o -c python/libsharp/libsharp.c
-	$(CL) -shared python/libsharp/libsharp.o -L$(LIBDIR) -lsharp -lfftpack -lc_utils `python-config --libs` -o $@
+	$(CC) $(DEBUG_CFLAGS) $(OPENMP_CFLAGS) $(PIC_CFLAGS) `python-config --cflags` -I$(INCDIR) -Ilibsharp -o python/libsharp/libsharp.o -c python/libsharp/libsharp.c
```

Comment:
Adding in `-Ilibsharp` here feels wrong, it will fail if you do `SHARP_TARGET=anotherconfig make <...>`. It is what the `-I$(INCDIR)` is supposed to do, didn't that work?